connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -232,10 +233,14 @@ public ConnectorStateInfo connectorStatus(String connName) {
 
     @Override
     public ConfigInfos validateConfigs(String connType, Map<String, String> connectorConfig) {
-        ConfigDef connectorConfigDef = ConnectorConfig.configDef();
-        List<ConfigValue> connectorConfigValues = connectorConfigDef.validate(connectorConfig);
-
         Connector connector = getConnector(connType);
+        ConfigDef connectorConfigDef;
+        if (connector instanceof SourceConnector) {
+            connectorConfigDef = ConnectorConfig.configDef();
```

Comment:
It looks like you changed the other instances, but shouldn't this also be `SourceConnectorConfig` for symmetry given we use `SinkConnectorConfig` below?